### ISSUE TITLE
Discriminator Generator imbalance

### ISSUE THREAD

**Issue by egesko** [egesko/NoiceGAN#1]
Currently, disciminator is outperforming the generator. I believe possible fix to this is by changing the loss function for the generator, which is currently the cross entropy of (1-y_real) and y_pred. This might be counterintuitive because it is hard for generator to be optimized to both decreasing the output of y_real, while increasing the output of every other discriminator neuron

![image](https://user-images.githubusercontent.com/49740123/168889930-138b86af-79bf-4b70-90bb-bd9a6888d9ec.png)

**Comment by egesko** [comment]
Another problem that might be causing this is the use of tanh as activation function at the end of generator model. Normally, tanh would've made sense since generator model usually tries to output realistic looking samples. But since we're not trying to do that, this seems pointless. I'll be normalizing the dataset differently ([0,1] instead of [-1,1]), and will be changing the final activation function of the generator.

**Comment by egesko** [comment]
Fixed the problem by having an extra output regularization parameter in the loss function. This parameter is tweaked to tweak the strength of generator model.
